discovery_engine_core/core/src/engine.rs

```diff
@@ -582,62 +583,104 @@ impl Engine {
         &mut self,
         history: Option<&[HistoricDocument]>,
         sources: &[WeightedSource],
-        reacted: &UserReacted,
-    ) -> Result<(), Error> {
+        reacted: UserReacted,
+    ) -> Result<Document, Error> {
```

Comment:
❓ instead of returning the whole document, wouldn't it be simpler to change `_changedDocsReporter.notifyChanged([document])` on the dart side such that the reporter only gets a list of document ids?

Reply:
It's part of the public interface consumed by the app.

The design is (ot least at some point it was):

```
   user reacts
     -> call engine with new reaction
     -> recv document change event 
     -> update display with the data received from the event
```
so this will brake the app if it is still implemented this way

Also we anyway need to fetch some of the values from the document from the DB and most code for passing the document from the engine to dart is there, too. So it's not that it adds a lot of overhead anyway.